File: Queue/codeforces_1859B.py

# Problem link ===>> https://codeforces.com/problemset/problem/1859/B

import sys
from heapq import heappush, heappop


def solve():
    q = int(input())

    overall_min = float('inf')
    min_pairs = []

    for _ in range(q):
        n = int(input())
        a = list(map(int, input().split()))
        pq = []  # Use a list for the priority queue

        for num in a:
            heappush(pq, -num)  # Store negative values for min-heap behavior
            if len(pq) > 2:
                heappop(pq)
            overall_min = min(overall_min, num)

        min_one = -heappop(pq)
        min_two = -heappop(pq)
        min_pairs.append((min_two, min_one))

    sum = 0
    current_min = float('inf')
    for min_two, min_one in min_pairs:
        sum += min_one
        current_min = min(current_min, min_one)

    sum += overall_min
    sum -= current_min
    print(sum)


def main():
    sys.stdin.readline()  # Ignore the first line (number of test cases)
    while True:
        try:
            solve()
        except EOFError:
            break


if __name__ == "__main__":
    main()

File: Queue/codeforces_1859B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1859/B
// submission link ===>> https://codeforces.com/contest/1859/submission/241197038

#include<bits/stdc++.h>
using namespace std;

void solve()
{
    int q;
    cin >> q;

    int overallMin = INT_MAX; // Use INT_MAX for simplicity
    vector<pair<int, int>> minPairs;

    for (int k = 0; k < q; k++)
    {
        int n;
        cin >> n;

        vector<int> a(n);
        priority_queue<int> pq;

        for (int i = 0; i < n; i++)
        {
            cin >> a[i];
            pq.push(a[i]);
            if (pq.size() > 2)
            {
                pq.pop();
            }
            overallMin = min(overallMin, a[i]);
        }

        int minOne = pq.top();
        pq.pop();
        int minTwo = pq.top();
        minPairs.push_back({minTwo, minOne});
    }

    long long int sum = 0;
    int currentMin = INT_MAX;
    for (const auto &[minTwo, minOne] : minPairs)
    {
        sum += minOne;
        currentMin = min(currentMin, minOne);
    }

    sum += overallMin;
    sum -= currentMin;
    cout << sum << '\n';
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int t = 1;
    cin >> t;
    while (t--)
    {
        solve();
    }

    return 0;
}
